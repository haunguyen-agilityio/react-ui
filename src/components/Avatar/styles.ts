// lib MUI
import { Avatar } from '@mui/material'
import { styled } from '@mui/system'

// theme
import theme from '../../themes'

export const StyledAvatar = styled(Avatar)({
  backgroundColor: '#21d083',
  fontWeight: 600,
  color: theme.palette.common.white,
})

export const SmallAvatar = styled(Avatar)({
  backgroundColor: '#aaaaaa',
})
