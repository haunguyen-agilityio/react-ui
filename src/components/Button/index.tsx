// libraries
import React, { ReactNode } from 'react'

// styles
import StyledButton from './styles'

interface IButtonProps {
  text: string
  onClick: () => void
  disable?: boolean
  color?: 'primary' | 'secondary'
  startIcon?: ReactNode
  endIcon?: ReactNode
}

const Button = ({
  text = 'Button',
  disable = false,
  color = 'primary',
  startIcon = false,
  endIcon = false,
  onClick = () => {},
}: IButtonProps) => {
  return (
    <StyledButton
      color={color}
      disabled={disable}
      variant='contained'
      startIcon={startIcon ? startIcon : false}
      endIcon={endIcon ? endIcon : false}
      onClick={onClick}
    >
      {text}
    </StyledButton>
  )
}

export default Button
