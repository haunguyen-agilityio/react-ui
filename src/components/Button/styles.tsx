// libraries
import { styled } from '@mui/system'

// components
import { Button } from '@mui/material'

// theme
import typography from '../../themes/typography'

const StyledButton = styled(Button)(({ theme }) => ({
  margin: theme.spacing(),
  padding: `${theme.spacing(1.7)} ${theme.spacing(3)}`,
  height: '38px',
  borderRadius: 0,
  boxShadow: 'none',
  textTransform: 'capitalize',
  fontSize: typography.h4.fontSize,
  color: theme.palette.common.white,
  '&:hover': {
    boxShadow: 'none',
  },
  '&.Mui-disabled': {
    color: theme.palette.common.black,
    backgroundColor: theme.palette.primary.main,
    opacity: 0.5,
  },
  '&.MuiButton-containedSecondary': {
    color: theme.palette.common.black,
    backgroundColor: '#eaeaea',
    border: '1px solid #cbcbcb',
    '&:hover': {
      backgroundColor: '#d0d0d0',
    },
    '&.Mui-disabled': {
      backgroundColor: '#eaeaea',
    },
  },
  '.MuiSvgIcon-root': {
    width: '12px',
    height: '12px',
  },
}))

export default StyledButton
