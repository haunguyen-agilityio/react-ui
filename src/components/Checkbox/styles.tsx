// libraries
import { Checkbox, CheckboxProps, FormControlLabel } from '@mui/material'
import { styled } from '@mui/system'

export const StyledFormControlLabel = styled(FormControlLabel)(({ theme }) => ({
  '.MuiFormControlLabel-label.Mui-disabled': {
    color: theme.palette.common.black,
  },
}))

interface ICheckboxProps extends CheckboxProps {
  isError?: boolean
}

export const StyledCheckbox = styled((props: ICheckboxProps) => <Checkbox {...props} />, {
  shouldForwardProp: (prop) => prop !== 'isError',
})(({ theme, isError }) => ({
  color: isError ? theme.palette.error.main : '#dedede',
  '&:hover': {
    backgroundColor: 'transparent',
  },
  '&.Mui-disabled': {
    opacity: 0.2,
  },
  '.MuiSvgIcon-root': {
    transform: 'scale(1.25)',
  },
  '&.Mui-checked': {
    color: theme.palette.primary.main,
  },
}))
