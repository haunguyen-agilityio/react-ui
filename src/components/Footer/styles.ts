// Libraries
import { styled } from '@mui/system'
import ListItemButton from '@mui/material/ListItemButton'

// Themes
import theme from '../../themes'

export const StyledListItemButton = styled(ListItemButton)({
  textAlign: 'center',
  color: theme.palette.common.white,
  marginRight: theme.spacing(0.5),
  padding: theme.spacing(1.5, 0),
  '&:hover': {
    backgroundColor: theme.palette.grey[400],
    color: theme.palette.common.black,
  },
  '&.Mui-selected': {
    backgroundColor: theme.palette.grey[400],
    color: theme.palette.common.black,
  },
})

export const StyledFooter = styled('footer')({
  color: theme.palette.common.white,
  backgroundColor: theme.palette.grey[500],
  padding: theme.spacing(0, 4.2),
})
