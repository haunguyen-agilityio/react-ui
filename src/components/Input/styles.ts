// Libraries
import { styled } from '@mui/system'
import TextField from '@mui/material/TextField'
import FormHelperText from '@mui/material/FormHelperText'

// Themes
import theme from '@themes'
import typography from '@themes/typography'

export const InputWrapper = styled('div')({
  padding: theme.spacing(2),
})

export const StyledFormHelperText = styled(FormHelperText)({
  color: theme.palette.error.main,
  fontSize: typography.body1.fontSize,
})

export const StyledTextField = styled(TextField)((props) => ({
  '& .MuiInputBase-input': {
    color: props.error ? theme.palette.error.main : theme.palette.text.primary,
  },
  ' .MuiOutlinedInput-root': {
    '& fieldset': {
      borderRadius: 0,
      padding: theme.spacing(0),
      borderColor: props.error ? theme.palette.error.main : theme.palette.grey[400],
      '&.MuiOutlinedInput-notchedOutline': {
        borderColor: props.error ? theme.palette.error.main : theme.palette.grey[400],
      },
    },
    '&:hover fieldset': {
      border: props.error
        ? `1px solid ${theme.palette.error.main}`
        : `1px solid ${theme.palette.grey[400]}`,
    },
    '&.Mui-focused fieldset': {
      borderColor: theme.palette.grey[400],
      border: props.error
        ? `1px solid ${theme.palette.error.main}`
        : `1px solid ${theme.palette.grey[400]}`,
    },
  },
}))
