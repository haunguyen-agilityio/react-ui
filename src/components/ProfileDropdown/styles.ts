// Libraries
import { styled } from '@mui/system'
import Accordion from '@mui/material/Accordion'
import AccordionSummary from '@mui/material/AccordionSummary'
import { MenuItem, MenuList, Link } from '@mui/material'

// Themes
import theme from '@themes'

export const DropdownStyles = styled('div')({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  width: '100%',
})

export const StyledAccordion = styled(Accordion)({
  width: theme.spacing(26.8),
  padding: theme.spacing(1.6),
  fontSize: theme.spacing(1.6),
})

export const StyledAccordionSummary = styled(AccordionSummary)({
  border: `${theme.spacing(0.1)} solid ${theme.palette.grey[300]}`,
})

export const StyledMenuItem = styled(MenuItem)({
  borderTop: `${theme.spacing(0.1)} solid ${theme.palette.grey[300]}`,
  padding: theme.spacing(1.6),
})

export const StyledMenuList = styled(MenuList)({
  width: theme.spacing(30),
  padding: theme.spacing(0),
})

export const StyledLink = styled(Link)({
  textDecoration: 'none',
  color: theme.palette.text.primary,
  fontSize: theme.spacing(1.6),
})
