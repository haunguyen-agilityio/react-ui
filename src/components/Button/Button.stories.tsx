// libraries
import { ComponentMeta, ComponentStory } from '@storybook/react'
import { ThemeProvider } from '@mui/material/styles'

// components
import Button from '@components/Button'

// theme
import theme from '@themes'

export default {
  component: Button,
  title: 'Components/Button',
} as ComponentMeta<typeof Button>

const Template: ComponentStory<typeof Button> = (args) => (
  <ThemeProvider theme={theme}>
    <Button {...args} />
  </ThemeProvider>
)

export const Primary = Template.bind({})
Primary.args = {
  text: 'Button',
  color: 'primary',
}

export const DisabledPrimary = Template.bind({})
DisabledPrimary.args = {
  text: 'Button',
  color: 'primary',
  disable: true,
}

export const Secondary = Template.bind({})
Secondary.args = {
  text: 'Button',
  color: 'secondary',
}

export const DisabledSecondary = Template.bind({})
DisabledSecondary.args = {
  text: 'Button',
  color: 'secondary',
  disable: true,
}
