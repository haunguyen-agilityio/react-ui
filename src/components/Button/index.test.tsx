// libraries
import { render, fireEvent } from '@testing-library/react'
import * as renderer from 'react-test-renderer'
import { ThemeProvider } from '@mui/material/styles'

// components
import Button from '../Button'

// theme
import theme from '../../themes'

describe('Button component', () => {
  const onClick = jest.fn()

  const ButtonComponent = ({ disable = false }) => (
    <ThemeProvider theme={theme}>
      <Button text='Button' onClick={onClick} disable={disable} />
    </ThemeProvider>
  )

  test('It should matches snapshot', () => {
    const button = renderer.create(<ButtonComponent />).toJSON()
    expect(button).toMatchSnapshot()
  })

  test('It should render correctly', () => {
    const { getByText } = render(<ButtonComponent />)
    expect(getByText('Button')).toBeInTheDocument()
  })

  test('On click function should be called once when click button', () => {
    const { getByText } = render(<ButtonComponent />)
    fireEvent.click(getByText('Button'))
    expect(onClick).toHaveBeenCalledTimes(1)
  })

  test('Button should be disabled', () => {
    const { getByText } = render(<ButtonComponent disable={true} />)
    expect(getByText('Button')).toBeDisabled()
  })
})
