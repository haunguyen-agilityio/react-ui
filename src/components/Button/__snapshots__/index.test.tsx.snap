// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component It should matches snapshot 1`] = `
<button
  className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-7koprg-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[MockFunction]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Button
</button>
`;
